mobot/mobot_client/payments.py

```diff
@@ -0,0 +1,186 @@
+        if len(list_of_txos) > 1:
+            raise ValueError(
+                "Found more than one txout for this chat bot-initiated transaction."
+            )
+
+        return list_of_txos[0]["txo_id_hex"]
+
+    def send_payment_receipt(self, source, tx_proposal):
+        receiver_receipt = self.create_receiver_receipt(tx_proposal)
+        receiver_receipt = mc.utility.full_service_receipt_to_b64_receipt(
+            receiver_receipt
+        )
+        resp = self.signal.send_payment_receipt(source, receiver_receipt, "Refund")
+        print("Send receipt", receiver_receipt, "to", source, ":", resp)
+
+    def create_receiver_receipt(self, tx_proposal):
+        receiver_receipts = self.mcc.create_receiver_receipts(tx_proposal)
+        # I'm assuming there will only be one receiver receipt (not including change tx out).
+        if len(receiver_receipts) > 1:
+            raise ValueError(
+                "Found more than one txout for this chat bot-initiated transaction."
+            )
+        return receiver_receipts[0]
+
+    def get_unspent_pmob(self):
+        account_amount_response = self.mcc.get_balance_for_account(self.account_id)
+        unspent_pmob = int(account_amount_response["unspent_pmob"])
+        return unspent_pmob
+
+    def get_minimum_fee_pmob(self):
+        return self.minimum_fee_pmob
+
+    def handle_item_payment(self, source, customer, amount_paid_mob, drop_session):
+        item_cost_mob = mc.pmob2mob(drop_session.drop.item.price_in_pmob)
+
+        if amount_paid_mob < item_cost_mob:
+            if mc.mob2pmob(amount_paid_mob) > self.minimum_fee_pmob:
+                self.messenger.log_and_send_message(
+                    customer,
+                    source,
+                    f"Not enough MOB, sending back {amount_paid_mob.normalize()} (minus network fees)",
+                )
+                self.send_mob_to_customer(customer, source, amount_paid_mob, False)
+            else:
+                self.messenger.log_and_send_message(
+                    customer,
+                    source,
+                    "Not enough MOB, unable to refund since it is less than the network fee",
+                )
+            return
+
+        if (
+                mc.mob2pmob(amount_paid_mob)
+                > mc.mob2pmob(item_cost_mob) + self.minimum_fee_pmob
+        ):
+            excess = amount_paid_mob - item_cost_mob
+            self.messenger.log_and_send_message(
+                customer,
+                source,
+                f"Sent too much MOB, sending back excess of {excess.normalize()} (minus network fees)",
+            )
+            self.send_mob_to_customer(customer, source, excess, False)
+
+        available_options = []
+        skus = Sku.objects.filter(item=drop_session.drop.item)
+
+        for sku in skus:
+            number_ordered = Order.objects.filter(sku=sku).count()
+            if number_ordered < sku.quantity:
+                available_options.append(sku)
```

Comment:
Looks like this is duplicated in a bunch of places. The Django way to solve this is to implement a custom queryset/manager (see https://docs.djangoproject.com/en/3.2/topics/db/managers/) so that you could do `skus = Sku.objects.filter(item=drop_session.drop.item).available_for_ordering()`.